wifiphisher/common/extensions.py

```diff
@@ -0,0 +1,194 @@
+import os
+import importlib
+import constants
+import threading
+import collections
+import scapy.layers.dot11 as dot11
+import scapy.arch.linux as linux
+
+class ExtensionManager():
+    """
+    Extension Manager (EM) defines an API for modular 
+    architecture in Wifiphisher. 
+
+    All extensions that lie under "extensions" directory
+    and are also defined in EXTENSIONS constant are loaded
+    and leveraged by EM. Each extension can take advantage
+    of the second wireless card (the first is used for the
+    rogue AP), aka run in "Advanced mode".
+
+    Each extension needs to be defined as a class that has
+    the name of the filename in camelcase. For example,
+    deauth.py would have a Deauth() class. Currently, 
+    extensions need to provide the following methods:
+
+    * __init__(self, data): Basic initialization that
+      received a dictionary with data from the main engine.
+
+    * get_packet(self, pkt): Method to process individually
+      each packet captured from the second card (monitor
+      mode). 
+
+    * send_output(self): Method that returns in a list
+      of strings the entry logs that need to be output.
+    """
+
+    def __init__(self):
+        """
+        Init the EM object. 
+
+        :param self: An ExtensionManager object
+        :type self: ExtensionManager
+        :param interface: String interface object
+        :type interface: String
+        :param shared_data: Dictionary object
+        :type shared_data: Dictionary
+        :return: None
+        :rtype: None
+        """
+
+        self._extensions = []
+        self._interface = None
+        self._socket = None
+        self._should_continue = True
+        self._packets_to_send = [] 
+
+    def set_interface(self, interface):
+        """
+        Sets interface for EM.
+
+        :param interface: Interface name
+        :type interface: String
+        :return: None
+        :rtype: None
+        """
+
+        self._interface = interface
+        self._socket = linux.L2Socket(iface=self._interface)
+
+    def init_extensions(self, shared_data):
+        """
+        Init EM extensions. Should be run 
+        when all shared data has been gathered.
+
+        :param self: An ExtensionManager object
+        :type self: ExtensionManager 
+        :param shared_data: Dictionary object
+        :type shared_data: Dictionary
+        :return: None
+        :rtype: None
+        """
+
+        # Convert shared_data from dict to named tuple
+        shared_data = collections.namedtuple('GenericDict', \
+                      shared_data.keys())(**shared_data)
+        # Initialize all extensions with the shared data
+        for m in constants.EXTENSIONS:
```

Comment:
Here the definition of `constants.EXTENSIONS = ["deauth", "lure10"]`, this means that when user uses the deauth interface, we will start lure10 attack. Is this what you want? Since I remember will have `--lure10` option for users before to indicate whether they want to do this attack or not.